### PR TITLE
*: add a system variable auto_analyze_ratio

### PR DESCRIPTION
**Pull request by lamxTyler**
So we can disable it dynamically.
PTAL @coocood @winoros @fipped

### CODE REVIEW

File: statistics/update.go [hunk 1]
```diff
@@ -393,9 +390,15 @@ func needAnalyzeTable(tbl *Table, limit time.Duration) bool {
 	return true
 }
 
+const minAutoAnalyzeRatio = 0.3
+
 // HandleAutoAnalyze analyzes the newly created table or index.
 func (h *Handle) HandleAutoAnalyze(is infoschema.InfoSchema) error {
 	dbs := is.AllSchemaNames()
+	autoAnalyzeRatio := h.ctx.GetSessionVars().AutoAnalyzeRatio
```
Comment (shenli) [inline statistics/update.go]: Load the value of `AutoAnalyzeRatio` from mysql.system_variable table.

File: session/session.go [hunk 2]
```diff
@@ -1272,7 +1272,8 @@ const loadCommonGlobalVarsSQL = "select HIGH_PRIORITY * from mysql.global_variab
 	variable.TiDBIndexSerialScanConcurrency + quoteCommaQuote +
 	variable.TiDBHashJoinConcurrency + quoteCommaQuote +
 	variable.TiDBBackoffLockFast + quoteCommaQuote +
-	variable.TiDBDistSQLScanConcurrency + "')"
+	variable.TiDBDistSQLScanConcurrency + quoteCommaQuote +
+	variable.TiDBAutoAnalyzeRatio + "')"
```
Comment (shenli) [inline session/session.go]: Do not need to load this variable for every session.

File: statistics/update.go [hunk 3]
```diff
@@ -393,9 +391,26 @@ func needAnalyzeTable(tbl *Table, limit time.Duration) bool {
 	return true
 }
 
+const minAutoAnalyzeRatio = 0.3
+
 // HandleAutoAnalyze analyzes the newly created table or index.
 func (h *Handle) HandleAutoAnalyze(is infoschema.InfoSchema) error {
 	dbs := is.AllSchemaNames()
+	sql := fmt.Sprintf("select variable_value from mysql.global_variables where variable_name = '%s'", variable.TiDBAutoAnalyzeRatio)
```
Comment (coocood) [inline statistics/update.go]: How about we extract those line into a function?

**Comment by coocood** [comment]
LGTM

**Pull request merged**